ci/dev-env-install.sh

```diff
@@ -33,4 +33,17 @@ source dev-env/lib/ensure-nix
 export NIX_CONF_DIR=$PWD/dev-env/etc
 
 step "Building dev-env dependencies"
-nix-build nix -A tools -A cached
+
+# Nix cache downloads can sometimes be flaky and end with "unexpected end-of-file" so we
+# repeat this a few times. There does not seem to be an option that we can pass to nix
+# to make it retry itself.
+for i in `seq 10`; do
+    nix-build nix -A tools -A cached 2>&1 | tee nix_log
+    # It should be in the last line but let’s use the last 3 and wildcards
+    # to be robust against slight changes.
+    if [[ $(tail -n 3 nix_log) == *"unexpected end-of-file"* ]]; then
```

Comment:
Why tail the log? Why not just retry 10 times unconditionally. I can imagine lots of flakey errors (connection not available for example), or someone changing the error message. I can't see why trying 10 times on failure would be harmful.

Reply:
Not harmful but I’d like to know which errors we get and so far I haven’t seen other flaky errors. So I’d prefer an explicit list at least while the number of errors is small (let’s say < 5)